examples/3d-tiles/app.js

```diff
@@ -102,51 +128,72 @@ export class App extends PureComponent {
     this.setState(
       {
-        pointsCount: featureTableJson.POINTS_LENGTH,
-        positions,
-        colors,
-        constantRGBA,
-        normals,
+        tile,
         viewState
       },
       this._rotateCamera
     );
   }
 
+  _getColor({index, data, target}) {
+    const {colors, isRGB565, constantRGBA, batchTable, batchIds} = this.state.tile;
+    if (colors) {
+      if (isRGB565) {
+        const color16 = data.colors.value[index];
+        const r5 = color16 & 31;
+        const g6 = (color16 >> 5) & 63;
+        const b5 = (color16 >> 11) & 31;
+        target[0] = Math.round((r5 * 255) / 32);
+        target[1] = Math.round((g6 * 255) / 64);
+        target[2] = Math.round((b5 * 255) / 32);
+        target[3] = 255;
+      } else {
+        target[0] = data.colors.value[index * 3];
+        target[1] = data.colors.value[index * 3 + 1];
+        target[2] = data.colors.value[index * 3 + 2];
+        target[3] = data.colors.size === 4 ? data.colors[index * 3 + 4] : 255;
+      }
+
+      return target;
+    }
+
+    if (constantRGBA) {
+      return constantRGBA;
+    }
+
+    if (batchIds) {
+      const batchId = batchIds[index];
+      // TODO figure out what is `dimensions` used for
+      const dimensions = batchTable.getProperty(batchId, 'dimensions');
+      const color = dimensions.map(d => d * 255);
+      return [...color, 255];
+    }
+
+    return [255, 255, 255];
+  }
+
   _renderLayers() {
-    const {pointsCount, positions, colors, constantRGBA, normals} = this.state;
+    if (!this.state.tile) {
+      return null;
+    }
+    const {pointsCount, positions, colors, normals} = this.state.tile;
 
     return (
       positions &&
       new PointCloudLayer({
         data: {
-          positions: {value: positions, size: 3},
-          colors: {value: positions, size: 4},
+          colors: {value: colors, size: 4},
           normals: {value: positions, size: 3},
           length: positions.length / 3
         },
         id: '3d-point-cloud-layer',
         coordinateSystem: COORDINATE_SYSTEM.IDENTITY,
         numInstances: pointsCount,
-        getPosition: (object, {index, data, target}) => {
-          target[0] = data.positions.value[index * 3];
-          target[1] = data.positions.value[index * 3 + 1];
-          target[2] = data.positions.value[index * 3 + 2];
-          return target;
-        },
-        getColor: colors
-          ? (object, {index, data, target}) => {
-              target[0] = data.colors.value[index * 3];
-              target[1] = data.colors.value[index * 3 + 1];
-              target[2] = data.colors.value[index * 3 + 2];
-              target[3] = data.colors.size === 4 ? data.colors[index * 3 + 4] : 255;
-              return target;
-            }
-          : constantRGBA
-            ? constantRGBA
-            : [255, 255, 255],
+        instancePositions: positions,
+        getColor: (object, {index, data, target}) => this._getColor({index, data, target}),
```

Comment:
Maybe make the signature of _getColor same as accessor then just 

`getColor: this._getColor`?